frontend/src/streamlist/StreamsTable.jsx

```diff
@@ -1,62 +1,129 @@
 import React from "react";
-import {
-  Box,
-  Table,
-  TableBody,
-  TableContainer,
-  TableHead,
-  TableRow
-} from "@material-ui/core";
+import { Box, TableContainer, Typography } from "@material-ui/core";
 import PropTypes from "prop-types";
-import HeadCells from "./HeadCells";
-import SingleStreamRow from "./SingleStreamRow";
 
-export default class StreamsTable extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      streams: []
-    };
-    this.dataSource = props.dataSource;
-    this.handleStreamsChange = this.handleStreamsChange.bind(this);
-  }
+import {
+  ExpandLess,
+  ExpandMore,
+  ArrowDownward,
+  FirstPage,
+  LastPage,
+  ChevronRight,
+  ChevronLeft
+} from "@material-ui/icons";
 
-  componentDidMount() {
-    this.dataSource.getAllStreams(this.handleStreamsChange);
-  }
+import MaterialTable from "material-table";
 
-  handleStreamsChange(streams) {
-    this.setState({
-      streams
-    });
-  }
+import StatusIndicator from "../general/StatusIndicator";
+import ActionButtons from "./ActionButtons";
+import StreamInfo from "../model/StreamInfo";
+
+function getColumnInfo() {
+  return [
+    {
+      title: "ID",
+      field: "id"
+    },
+    {
+      title: "Date",
+      field: "date"
+    },
+    {
+      title: "Sender",
+      field: "sender.name"
+    },
+    {
+      title: "Receiver",
+      field: "receiver.name"
+    },
+    {
+      title: "Status",
+      field: "status",
+      render: (rowData) => <StatusIndicator status={rowData.status} />
+    },
+    {
+      title: "Type",
+      field: "type"
+    },
+    {
+      title: "Time Elapsed",
+      field: "time"
+    },
+    {
+      title: "Actions",
+      field: "action",
+      filtering: false,
+      sorting: false,
+      render: (rowData) => <ActionButtons streamId={rowData.id} />,
+      align: "center",
+      export: false
+    }
+  ];
+}
 
+function getDetailPanel() {
+  return [
+    {
+      icon: ExpandMore,
+      openIcon: ExpandLess,
+      tooltip: "Show Stream Details",
+      render: (rowData) => {
+        return (
+          <div className="lightestGrey">
+            <Typography variant="h6">Stream Details</Typography>
+          </div>
+        );
```

Comment:
Can we get rid of this smell? "rowData is defined but never used"

Reply:
Yup, should be able to get rid of it, good catch!